src/lib/repositories_v2/impl/order_repository_impl.py

```diff
@@ -0,0 +1,154 @@
+        )
+        return list(orders)
+
+    def delete_by_id(self, order_id, order):
+        self.session.query(Order).filter(Order.id == order_id).update(
+            {
+                Order.entity_status: Status.DELETED.value,
+                Order.updated_date: datetime.now(),
+                Order.updated_by: order.updated_by,
+            }
+        )
+        self.session.commit()
+
+    def update_by_id(self, order_id, order):
+        order_to_be_updated = (
+            self.session.query(Order).filter(Order.id == order_id).first()
+        )
+        order_to_be_updated.user_id = order.user_id or order_to_be_updated.user_id
+        order_to_be_updated.skill = order.skill or order_to_be_updated.skill
+        order_to_be_updated.updated_date = datetime.now()
+        order_to_be_updated.updated_by = order.updated_by
+        self.session.add(order_to_be_updated)
+        self.session.commit()
+
+    def get_orders_by_status(self, order_status, order_limit=None):
+        orders_by_status = (
+            self.session.query(Order)
+            .filter(Order.entity_status == Status.ACTIVE.value)
+            .filter(Order.status == order_status)
+            .limit(order_limit)
+            .all()
+        )
+
+        return orders_by_status
+
+    def get_order_ingredients_by_order_id(self, order_id):
+
+        product_ingredients_by_order_id = (
+            self.session.query(ProductIngredient)
+            .filter(
+                ProductIngredient.entity_status == Status.ACTIVE.value,
+                OrderDetail.entity_status == Status.ACTIVE.value,
+            )
+            .join(OrderDetail, ProductIngredient.product_id == OrderDetail.product_id)
+            .filter(OrderDetail.order_id == order_id)
+        )
+
+        return list(product_ingredients_by_order_id)
+
+    def get_validated_orders_map(self, orders_to_process):
+
+        case_query = case(
+            (
+                func.min(
+                    InventoryIngredient.quantity
+                    / ProductIngredient.quantity
+                    / OrderDetail.quantity
+                )
+                > 0,
+                True,
+            ),
+            else_=False,
+        )
+
+        validation_orders = (
+            self.session.query(OrderDetail.order_id, case_query)
+            .join(
+                ProductIngredient,
+                ProductIngredient.product_id == OrderDetail.product_id,
+            )
+            .join(
+                InventoryIngredient,
+                ProductIngredient.ingredient_id == InventoryIngredient.ingredient_id,
+            )
+            .filter(
+                OrderDetail.entity_status == Status.ACTIVE.value,
+                ProductIngredient.entity_status == Status.ACTIVE.value,
+                InventoryIngredient.entity_status == Status.ACTIVE.value,
+            )
+            .filter(OrderDetail.order_id.in_([order.id for order in orders_to_process]))
```

Comment:
Is confusing to have a mix of `filter` and `,` to perform `and` conditions, could do it one way if possible?

Reply:
Done.